### PR TITLE
Resolve manifest lists to manifests in remote.Image

### PR DESCRIPTION
**Pull request by jonjohnsonjr**
We currently rely on the registry to do fallback for us, but this is
unreliable:
* Some registries won't do it.
* This only works for amd64/linux

### CODE REVIEW

File: pkg/v1/remote/image.go [hunk 1]
```diff
@@ -283,3 +294,38 @@ func (r *remoteImage) LayerByDigest(h v1.Hash) (partial.CompressedLayer, error)
...
+	}
+	for _, childDesc := range index.Manifests {
+		p := v1.Platform{
+			Architecture: "amd64",
+			OS:           "linux",
+		}
+		if childDesc.Platform != nil {
+			p = *childDesc.Platform
+		}
+		if runtime.GOARCH == p.Architecture && runtime.GOOS == p.OS {
```
Comment (dlorenc) [inline pkg/v1/remote/image.go]: Would it make sense to make runtime.GOARCH and runtime.GOOS parameters here? I've had cases where I want to resolve things for a different platform (think crane digest on a macbook).

Comment (jonjohnsonjr) [reply pkg/v1/remote/image.go]: Yeah we probably want to expose something like `WithPlatform` in our image options, good thinking.

**Comment by dlorenc** [comment]
Looks like you're missing boilerplate in ././pkg/executor/push_test.go

File: pkg/v1/remote/image.go (hunk 1 above)
Comment (jonjohnsonjr) [reply pkg/v1/remote/image.go]: Changed this to default to amd64/linux to get the previous behavior (kind of). We can override that with `WithPlatform` wherever it makes sense.

**Pull request merged**